zerver/lib/actions.py

```diff
@@ -2491,6 +2502,50 @@ def update_flag(um, should_set, flag):
 
 # We use transaction.atomic to support select_for_update in the attachment codepath.
 @transaction.atomic
+def do_update_embeded_data(user_profile, message, content, rendered_content):
+    # type: (UserProfile, Message, Optional[text_type], Optional[text_type]) -> None
+    event = {
+        'type': 'update_message',
+        'sender': user_profile.email,
+        'message_id': message.id}  # type: Dict[str, Any]
+    changed_messages = [message]
+
+    ums = UserMessage.objects.filter(message=message.id)
+
+    if content is not None:
+        update_user_message_flags(message, ums)
+        message.content = content
+        message.rendered_content = rendered_content
+        message.rendered_content_version = bugdown_version
+        event["content"] = content
+        event["rendered_content"] = rendered_content
+
+    log_event(event)
+    message.save(update_fields=["content", "rendered_content"])
+
+    # Update the message as stored in the (deprecated) message
+    # cache (for shunting the message over to Tornado in the old
```

Comment:
This comment is wrong -- there no longer is a (deprecated) message cache, so just mention the `to_dict` caches.